go/libraries/doltcore/table/table_reader.go

```diff
@@ -34,29 +34,29 @@ type TableReader interface {
 	ReadRow(ctx context.Context) (row.Row, error)
 }
 
-// TableCloser is an interface for a table stream that can be closed to release resources
-type TableCloser interface {
+// Closer is an interface for a writer that can be closed to release resources
+type Closer interface {
 	// Close should release resources being held
 	Close(ctx context.Context) error
 }
 
-// TableReadCloser is an interface for reading rows from a table, that can be closed.
-type TableReadCloser interface {
-	TableReader
-	TableCloser
+// ReadCloser is an interface for reading rows from a table, that can be closed.
+type ReadCloser interface {
+	Reader
+	Closer
 }
 
 type SqlRowReader interface {
-	TableReadCloser
+	ReadCloser
 
 	ReadSqlRow(ctx context.Context) (sql.Row, error)
 }
 
-// SqlTableReader is a  TableReader that can read rows as sql.Row.
+// SqlTableReader is a Reader that can read rows as sql.Row.
 type SqlTableReader interface {
```

Comment:
(minor) Should `SqlTableReader` just embed an instance of `SqlRowReader` to pick up the `ReadSqlRow` method?

Reply:
Gonna leave it for now